lib/generators/shopify_app/install/install_generator.rb

```diff
@@ -7,17 +7,23 @@ class InstallGenerator < Rails::Generators::Base
       include Rails::Generators::Migration
       source_root File.expand_path('../templates', __FILE__)
 
+      class_option :application_name, type: :string
       class_option :api_key, type: :string, default: '<api_key>'
       class_option :secret, type: :string, default: '<secret>'
       class_option :scope, type: :string, default: 'read_orders, read_products'
       class_option :embedded, type: :string, default: 'true'
 
       def create_shopify_app_initializer
+        @application_name = options['application_name']
         @api_key = options['api_key']
         @secret = options['secret']
         @scope = options['scope']
 
-        template 'shopify_app.rb', 'config/initializers/shopify_app.rb'
+        unless @application_name.nil?
+          template 'shopify_app_with_appname.rb', 'config/initializers/shopify_app.rb'
```

Comment:
I tried inline conditional
`@application_name.present ? config.application_name = "<%= @application_name %>"` but that always left the query  segment visible.
I did not think of using a block method though.
I'll commit the above change now